portafly/portafly_api.yml

```diff
@@ -18,6 +18,27 @@ paths:
           $ref: '#/components/responses/developerAccountList'
       operationId: developerAccountList
       summary: Get developer account list
+    patch:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/DeveloperAccount'
+            examples:
+              Patch the state of 2 Dev Accounts:
+                value:
+                - id: 37
+                  state: active
+                - id: 18
+                  state: active
+        required: true
+      responses:
+        "204":
+          description: Resource updated correctly
+      summary: Partially updates developer accounts
```

Comment:
What does mean "Partially" it is not clear.
To my point of view update is changing some state so it can't be partial (like maybe it has changed or maybe not)

Reply:
Partial in the sense that you are not sending the entire record back, just the fields you want to update, not the entire record. https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH

Reply:
Not true. They just use an example and even explained it:

> A successful response is indicated by any 2xx status code.
>
> In the example below a 204 response code is used, because the response does not > carry a payload body. A 200 response could have contained a payload body.

Reply:
Why should we ever do this? It looks broken design to me

Reply:
Scenario: What happens if one record PATCHING is successful and the other not?

Reply:
I'm not saying it's the only way, just following practices and matching our use case since we have the changes in our state, if it's 204 back, we don't need any data from the server. And if one fails and the other not, you receive an error message back from the server, we show it that it failed... not big deal, maybe even in that case update the list.
SO if we want instead of having a 200 and return the records, should we return the records changed or all of them? If it fails, we should return the failed ones, the good ones, just an error message?

Reply:
You send only the field you want to update, then the backend returns the whole collection (updated). Is that what you mean @hallelujah ?

Reply:
I mean, by default you send all the resource.

First scenario:

- You update multiple resources (which is an anti pattern, but lets live with that for the sake of the argument)
- one of the resource fail to update, the other are updated. What is the correct status code if there is no body to tell you what really happens

Another scenario:

- all the resource updated but another process (outside of this React component) updated another attribute of this resource. Not sending back the resource in the body will lead for sure of a desync.

Reply:
https://blog.ploeh.dk/2013/04/30/rest-lesson-learned-avoid-204-responses/